addon/components/au-file-upload.gts

```diff
@@ -55,36 +89,42 @@ export default class AuFileUpload extends Component {
     return this.uploadErrorData.length > 0;
   }
 
-  @task
-  *upload(file) {
+  @action
+  upload(file: UploadFile): void | undefined {
+    this.uploadTask.perform(file);
+  }
+
+  uploadTask = task(async (file: UploadFile) => {
     this.resetErrors();
-    let uploadedFile = yield this.uploadFileTask.perform(file);
+    let uploadedFile: number | null = await this.uploadFileTask.perform(file);
```

Comment:
Hmm, I just noticed that this can be a `const` and eslint should complain about that. We're running into this issue:

https://github.com/ember-cli/eslint-plugin-ember/issues/2106

So we would have to temporarily add those rules to our eslint config until we switch to the flat config variant, or until eslint-plugin-ember does it for us.